xbmc/filesystem/VideoDatabaseDirectory/DirectoryNode.cpp

```diff
@@ -50,14 +50,14 @@ using namespace XFILE::VIDEODATABASEDIRECTORY;
 //  Constructor is protected use ParseURL()
 CDirectoryNode::CDirectoryNode(NODE_TYPE Type, const std::string& strName, CDirectoryNode* pParent)
 {
-  m_Type=Type;
-  m_strName=strName;
-  m_pParent=pParent;
+  m_Type = Type;
+  m_strName = strName;
+  m_pParent = pParent;
 }
 
 CDirectoryNode::~CDirectoryNode()
 {
-  delete m_pParent;
+  delete m_pParent, m_pParent = nullptr;
```

Comment:
I could use SAFE_DELETE(p), in system.h :)
That gives me the same 'one line of clean code' look but including system.h all over is rather painful. Humm, maybe I'll pull those out into their own header, system.h is rather a mess anyway.